kernel/sched.c

```diff
@@ -303,12 +303,12 @@ static ALWAYS_INLINE struct k_thread *next_up(void)
 		if (!should_preempt(thread, _current_cpu->swap_ok)) {
 			thread = _current;
 		}
-	}
 
-	/* Put _current back into the queue */
-	if (thread != _current && active &&
-		!z_is_idle_thread_object(_current) && !queued) {
-		queue_thread(&_kernel.ready_q.runq, _current);
+		/* Put _current back into the queue */
+		if (thread != _current &&
+		    !z_is_idle_thread_object(_current) && !queued) {
+			queue_thread(&_kernel.ready_q.runq, _current);
+		}
```

Comment:
Not sure I buy the aesthetics here.  The point to having that in a separate block at the top level is to clarify that it's a separate (and really important) operation where the current thread is being re-added to the run queue having been preempted.  Obviously the compiler is going to generate the same code either way, but honestly I like this the way it is.   Please remove.